src/main/java/command/order/DeleteOrderCommand.java

```diff
@@ -59,20 +42,60 @@ public void execute() {
         assert orderId <= Order.getOrderCount() : "order Id should not exceed max order count";
 
         for (Medicine medicine : medicines) {
+
             if (!(medicine instanceof Order)) {
                 continue;
             }
+
             Order order = (Order) medicine;
+
             if (order.getOrderId() == orderId) {
                 medicines.remove(order);
                 logger.log(Level.INFO, "Order id found and deleted");
                 break;
             }
         }
+
         ui.print("Order deleted for Order ID " + orderId);
         Storage storage = Storage.getInstance();
         storage.saveData(medicines);
         logger.log(Level.INFO, "Successful deletion of order");
+
+    }
+
+    /**
+     * Check if parameters values for Order are valid and if Order ID exist.
+     *
+     * @param ui Reference to the UI object to print messages.
+     * @param medicines Arraylist of medicines
+     * @param orderIdToDelete Order ID to delete.
+     * @return Boolean Value indicating if parameters values for Order are valid and Order ID exist.
+     */
+    private boolean isValidOrderParameters(Ui ui, ArrayList<Medicine> medicines, String orderIdToDelete) {
+
+        OrderValidator orderValidator = new OrderValidator();
+
+        String[] requiredParameters = {CommandParameters.ID};
+        String[] optionalParameters = {};
+
+        boolean isInvalidParameter = orderValidator.containsInvalidParameters(ui, parameters, requiredParameters,
+                optionalParameters, CommandSyntax.DELETE_ORDER_COMMAND, true);
+
+        if (isInvalidParameter) {
+            logger.log(Level.WARNING, "Invalid parameter is specified by user");
+            logger.log(Level.INFO, "Unsuccessful deletion of order");
+            return true;
+        }
+
+        boolean isValidOrderId = orderValidator.isValidOrderId(ui, orderIdToDelete, medicines);
+
+        if (!isValidOrderId) {
+            logger.log(Level.WARNING, "Invalid order id is specified by user");
+            logger.log(Level.INFO, "Unsuccessful deletion of order");
+            return true;
+        }
```

Comment:
Is this part necessary? should be caught in `containsInvalidParameters()` in line 81?

Reply:
Required to check if the order id provided by user exists